### PR TITLE
[ci] Use just filename for artifact upload and exclude the build dir prefix

### PR DESCRIPTION
**Pull request by brianseeders**
Artifact uploads are failing to be linked in the buildscan on Windows.

This is because the artifact name ends up having a directory in it, and thus a slash, and Buildkite flips the slash on Windows. So, the name of the artifact ends up being different between Linux and Windows.

We don't need the directory name as part of the artifact, so I'm just removing it in this PR, rather than handling different OSes differently.

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java [hunk 1]
```diff
@@ -142,10 +142,16 @@ public void execute(BuildFinishedFlowAction.Parameters parameters) throws FileNo
             uploadFile.getParentFile().mkdirs();
             createBuildArchiveTar(parameters.getFilteredFiles().get(), parameters.getProjectDir().get(), uploadFile);
             if (uploadFile.exists() && "true".equals(System.getenv("BUILDKITE"))) {
-                String uploadFilePath = "build/" + uploadFile.getName();
+                String uploadFilePath = uploadFile.getName();
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java:145]: Is there a reason we can't just pass the absolute path to `buildkite-agent artifact upload`?

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java [hunk 2]
```diff
@@ -142,10 +142,16 @@ public void execute(BuildFinishedFlowAction.Parameters parameters) throws FileNo
...
                 try {
                     System.out.println("Uploading buildkite artifact: " + uploadFilePath + "...");
-                    new ProcessBuilder("buildkite-agent", "artifact", "upload", uploadFilePath).start().waitFor();
+                    ProcessBuilder pb = new ProcessBuilder("buildkite-agent", "artifact", "upload", uploadFilePath);
+                    // If we don't switch to the build directory first, the uploaded file will have a `build/` prefix
+                    // Buildkite will flip the `/` to a `\` at upload time on Windows, which will make the search command below fail
+                    // So, if you change this such that the artifact will have a slash/directory in it, you'll need to update the logic
+                    // below as well
+                    pb.directory(new File("./build"));
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java:153]: Using relative paths with `new File()` is predictable. Can't we just use `uploadFile.getParentFile()` here?

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java (hunk 1 above)
Comment (brianseeders) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java:145]: I'm actually not sure what happens when you use the absolute paths, and the docs don't seem to mention it. I bet the entire path becomes part of the artifact name.

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java (hunk 2 above)
Comment (brianseeders) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java:153]: Yep, that's better, thanks

**Review by mark-vieira — APPROVED** [review]

**Comment by brianseeders** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.13|[](https://github.com/elastic/elasticsearch/pull/107368)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**